### PR TITLE
[datalake] Fix path rename operations

### LINKED ISSUE #651: File rename doesn't work in Storage Datalake API
Cause seems to be wrong header name here: https://github.com/Azure/azure-sdk-for-rust/blob/967b9bdc3388c6f305836abd3836b4430b7f4fc4/sdk/storage_datalake/src/request_options.rs#L201

### PR DESCRIPTION
**Pull request by roeap**
closes #651

@thovoll - here is the fix for the rename operation, hopefully unblocking you on the delta-rs side :).

To fix the rename operation a few things needed changing:
- header keys for `x-ms-rename-source` was wrong
- header value for  `x-ms-rename-source` should be url encoded
- resource query must not be set on rename operation

When writing some tests, I also noticed that parsing the properties header would fail, if no custom properties are set on a file or directory - thus I allowed for an empty (!= missing) property header in the response.

### CODE REVIEW

**Comment by thovoll** [comment]
Thanks @roeap, also for looking deeper and fixing a couple of other issues!

File: sdk/storage_datalake/src/request_options.rs [hunk 1]
```diff
@@ -197,10 +199,10 @@ impl AddAsHeader for RenameSource {
         &self,
         request: &mut azure_core::Request,
     ) -> Result<(), azure_core::HttpHeaderError> {
-        request.headers_mut().append(
-            http::header::CONTENT_LANGUAGE,
-            http::HeaderValue::from_str(&self.0)?,
-        );
+        let encoded: String = byte_serialize(self.0.as_bytes()).collect();
```
[the remark this answers is not in the document]
Comment (thovoll) [reply sdk/storage_datalake/src/request_options.rs:202]: Also, maybe add a test that would fail if the URL encoding is missing.

**Review by thovoll — CHANGES_REQUESTED** [review]
Looks good so far, almost there!

File: sdk/storage_datalake/examples/file_rename.rs [hunk 2]
```diff
@@ -43,8 +43,9 @@ async fn main() -> Result<(), Box<dyn Error + Send + Sync>> {
     );
 
     println!("renaming file '{}' to '{}'...", file_path1, file_path2);
-    let rename_file_response = file_client1.rename(file_path2).into_future().await?;
-    println!("rename file response == {:?}\n", rename_file_response);
+    let file_client3 = file_client1.rename(file_path2).into_future().await?;
+    let renamed_file_properties = file_client3.get_properties().into_future().await?;
+    println!("renamed file properties == {:?}\n", renamed_file_properties);
```
[the remark this answers is not in the document]
Comment (roeap) [reply sdk/storage_datalake/examples/file_rename.rs:48]: added some code here - in the tests we already do this in the tests. Also left a comment that this should be replaced by a `exists` function.

File: sdk/storage_datalake/src/properties.rs [hunk 3]
```diff
@@ -76,6 +76,14 @@ impl TryFrom<&HeaderMap> for Properties {
     fn try_from(headers: &HeaderMap) -> Result<Self, Self::Error> {
         let mut properties = Self::new();
 
+        let header_value = headers
+            .get(HEADER)
+            .ok_or_else(|| crate::Error::HeaderNotFound(HEADER.to_owned()))?
+            .to_str()?;
+        if header_value.is_empty() {
```
[the remark this answers is not in the document]
Comment (roeap) [reply sdk/storage_datalake/src/properties.rs:83]: Added a check in the rename test that queries properties on a file with no properties set. This is when this fails.

File: sdk/storage_datalake/src/request_options.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (roeap) [reply sdk/storage_datalake/src/request_options.rs]: I think the rename operation will fail - i.e. source does not get removed - when this is the case.

**Review by thovoll — APPROVED** [review]

**Pull request merged**